src/compiler/transformers/module/module.ts

```diff
@@ -846,6 +868,59 @@ namespace ts {
                 || node;
         }
 
+        function substituteBinaryExpression(node: BinaryExpression): Expression {
+            const left = node.left;
+            // If the left-hand-side of the binaryExpression is an identifier and its is export through export Specifier
+            if (isIdentifier(left) && isAssignmentOperator(node.operatorToken.kind)) {
+               const sourceFile = getSourceFileOfNode(node);

```

Comment:
`getSourceFileOfNode` depends on parent pointers, which may not exist. If you need to know the source file, keep track of it during the call to `onEmitNode`.